hikey_debian.mk

```diff
@@ -0,0 +1,408 @@
+	@xdg-open https://builds.96boards.org/snapshots/hikey/linaro/debian/410/hikey-jessie_alip_20160225-410.emmc.img.gz
+endif
+endif
+
+.PHONY: system-cleaner
+system-cleaner:
+	rm -f $(SYSTEM_IMG)
+
+################################################################################
+# l-loader
+################################################################################
+lloader: arm-tf
+	$(MAKE) -C $(LLOADER_PATH) BL1=$(ARM_TF_PATH)/build/hikey/$(ARM_TF_BUILD)/bl1.bin CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)" PTABLE_LST=linux-4g
+
+.PHONY: lloader-clean
+lloader-clean:
+	$(MAKE) -C $(LLOADER_PATH) clean
+
+################################################################################
+# nvme image
+################################################################################
+.PHONY: nvme
+nvme: prepare
+ifeq ("$(wildcard $(NVME_IMG))","")
+	wget https://builds.96boards.org/releases/hikey/linaro/binaries/latest/nvme.img -O $(NVME_IMG)
+endif
+
+.PHONY: nvme-cleaner
+nvme-cleaner:
+	rm -f $(NVME_IMG)
+
+################################################################################
+# Debian package
+################################################################################
+define CONTROL_TEXT
+Package: op-tee
+Version: $(OPTEE_PKG_VERSION)
+Section: base
+Priority: optional
+Architecture: arm64
+Depends:
+Maintainer: Joakim Bech <joakim.bech@linaro.org>
+Description: OP-TEE client binaries, test program and Trusted Applications
+ Package contains tee-supplicant, libtee.so, xtest and a set of
+ Trusted Applications.
+ NOTE! This package should only be used for testing and development.
+endef
+
+export CONTROL_TEXT
+
+.PHONY: deb
+deb: xtest optee-client
+	@mkdir -p $(DEBPKG_BIN_PATH) && cd $(DEBPKG_BIN_PATH) && \
+		cp -f $(OPTEE_CLIENT_EXPORT)/bin/tee-supplicant . && \
+		cp -f $(OPTEE_TEST_OUT_PATH)/xtest/xtest .
+
+	@mkdir -p $(DEBPKG_LIB_PATH) && cd $(DEBPKG_LIB_PATH) && \
+		cp $(OPTEE_CLIENT_EXPORT)/lib/libtee* .
+
+	@mkdir -p $(DEBPKG_TA_PATH) && cd $(DEBPKG_TA_PATH) && \
+		find $(OPTEE_TEST_OUT_PATH)/ta -name "*.ta" -exec cp {} . \;
+
+	@mkdir -p $(DEBPKG_CONTROL_PATH)
+	@echo "$$CONTROL_TEXT" > $(DEBPKG_CONTROL_PATH)/control
+	@cd $(OUT_PATH) && dpkg-deb --build optee_$(OPTEE_PKG_VERSION)
+
+################################################################################
+# Send built files to the host, note this require that the IP corresponds to
+# the device. One can run:
+#   IP=111.222.333.444 make send
+# If you don't want to edit the makefile itself.
+################################################################################
+.PHONY: send
+send:
+	@tar czf - $(shell cd $(OUT_PATH) && echo $(OUT_PATH)/*.deb && echo $(ROOT)/linux-image-*.deb) | ssh linaro@$(IP) "cd /tmp; tar xvzf -"
+	@echo "Files has been sent to $$IP/tmp/ and $$IP/tmp/out"
+	@echo "On the device, run:"
+	@echo " dpkg --force-all -i /tmp/out/optee_$(OPTEE_PKG_VERSION).deb"
+	@echo " dpkg --force-all -i /tmp/linux-image-*.deb"
```

Comment:
Any thoughts for choosing this approach over mounting the sys img and just copying the required files over locally? Then it's just flashing instead of having to deal with IPs, i.e. flashing, booting the board, configuring wifi and rebooting, before we can finally send?